packages/connect-webextension/src/contentScript.ts

```diff
@@ -1,81 +1,92 @@
 import { WindowServiceWorkerChannel } from '@trezor/connect-web/src/channels/window-serviceworker';
 import { POPUP } from '@trezor/connect/src/events/popup';
 
-/**
- * communication between service worker and both webextension and popup manager
- */
-const channel = new WindowServiceWorkerChannel({
-    name: 'trezor-connect',
-    channel: {
-        here: '@trezor/connect-content-script',
-        peer: '@trezor/connect-webextension',
-    },
-});
-
-/**
- * messages that were sent before the channel was initialized
- */
-const messagesQueue: any[] = [];
-let channelReady = false;
-
-/**
- * Firefox enforces some restrictions on the content script that force us to use clones of objects when passing them between the content script and the background script
- */
-function clone(obj: any) {
-    return JSON.parse(JSON.stringify(obj));
-}
-
-/*
- * Passing messages from popup to service worker
- */
-window.addEventListener('message', event => {
-    if (
-        event.data?.channel?.here === '@trezor/connect-webextension' ||
-        event.data?.type === POPUP.CONTENT_SCRIPT_LOADED
-    ) {
+function trezorContentScript() {
```

Comment:
don't we want to introduce in some separate versioning of content script?

lets say do something like:

```
CONTENT_SCRIPT_VERSION = 1
```

then we pass another param in query string such as SUPPORTED_CONTENT_SCRIPT_VERSION = 2 and this way we are able to detect version incompatibility between these elements and at least print an error to console?

Reply:
Good idea, I was also thinking about something like this, we don't necessarily need it for this change since it's still compatible, but it might be useful in the future. 
This should only happen in situations when the extension authors updates the NPM package, has the content script copied manually and doesn't update it.